clients/fides-js/src/fides-tcf.ts

```diff
@@ -80,29 +103,110 @@ declare global {
 // eslint-disable-next-line no-underscore-dangle,@typescript-eslint/naming-convention
 let _Fides: Fides;
 
+/** Helper function to determine the initial value of a TCF object */
+const getInitialPreference = (
+  tcfObject:
+    | TCFFeatureRecord
+    | TCFVendorConsentRecord
+    | TCFVendorLegitimateInterestsRecord
+): UserConsentPreference => {
+  if (tcfObject.current_preference) {
+    return tcfObject.current_preference;
+  }
+  return tcfObject.default_preference ?? UserConsentPreference.OPT_OUT;
+};
+
 const updateCookie = async (
   oldCookie: FidesCookie,
-  experience: PrivacyExperience
+  experience: PrivacyExperience,
+  debug?: boolean,
+  tc_str_override_set?: boolean
 ): Promise<FidesCookie> => {
-  // First check if the user has never consented before
+  // ignore server-side prefs if either user has no prefs to override, or TC str override is set
+  if (tc_str_override_set) {
+    return oldCookie;
+  }
   if (!hasSavedTcfPreferences(experience)) {
     return { ...oldCookie, tc_string: "" };
   }
 
-  // Usually at this point, we'd look at the Experience from the backend and update
-  // the user's browser cookie to match the preferences in the Experience. However,
-  // TCF requires pre-fetching an experience. A prefetch'd experience will never have user
-  // specific consents. We rely on the cookie to fill those in. Therefore, we do nothing
-  // here, since the cookie is the source of truth, not the backend Experience.
-  return oldCookie;
+  const tcSavePrefs: TcfSavePreferences = {};
+  const enabledIds: any | EnabledIds = {
+    purposesConsent: [],
+    purposesLegint: [],
+    specialPurposes: [],
+    features: [],
+    specialFeatures: [],
+    vendorsConsent: [],
+    vendorsLegint: [],
+  };
+
+  TCF_KEY_MAP.forEach(({ experienceKey, cookieKey, enabledIdsKey }) => {
+    if (experienceKey) {
+      tcSavePrefs[cookieKey] = [];
+      experience[experienceKey]?.forEach((record) => {
+        const pref: UserConsentPreference = getInitialPreference(record);
+        // map experience to tcSavePrefs (same as cookie keys)
+        tcSavePrefs[cookieKey]?.push({
+          // @ts-ignore
```

Comment:
hmmm this part of the code starts to have a bit of overlap with some of the initialization logic that `TcfOverlay` does. we might want to look at it as a whole and extract some common functions out at some point, but we don't have to do that now

Reply:
yeah totally. It's tricky bc for `fides-tcf.ts` we don't have `enabledIds` upfront like we do for `TcfOverlay`. So we'd have to loop through the `experience` and generate those first. 

But we actually do use a core shared method down the line `transformTcfPreferencesToCookieKeys` which is something...